API provides bookmark functionality for user

piavW opened the pull request:
[PT-link](https://www.pivotaltracker.com/story/show/169667623)

## PR Content
```
As a user of the website
In order to find trails I'm interested in again
I'd like to able to bookmark a specific trail
```

## This PR Contains
- Associations between User, Bookmark and Trail

## What we've learned
- Deeper understanding of Rails 
- Deeper knowledge of Active Record Associations

piavW committed:
Creates test for bookmark model.

piavW committed:
Updates bookmark specs with associations

- Creates bookmark factory, adds associations
- Updates bookmark model with associations

Co-Authored-By: Alex Saxena, <saxenaalex@gmail.com>

piavW committed:
Creates & configures Bookmark controller & model

- Creates controller and model
- Creates belongs_to association in Bookmark model
- Adds association has_many through in user-model and trail-model
- Migrates Bookmark migration-file to schema
- Defines build action in Bookmark controller

Co-Authored-By: Alex Saxena, <saxenaalex@gmail.com>

DIzosimov commented on app/controllers/bookmark_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class BookmarkController < ApplicationController
+  def build 
+
```
something is missing here...

DIzosimov replied:
im not quite sure what though..

AlexSaxena committed:
Moves controller to V1, Creates request spec-file.

piavW committed:
Creates request tests for bookmark functionality

Co-Authored-By: Alex Saxena, <saxenaalex@gmail.com>

piavW committed:
Creates User controller & updates actions

- Updates bookmark controller with index and create actions
- Adds show action for user controller
- Updates routes for user and bookmark actions

Co-Authored-By: Alex Saxena, <saxenaalex@gmail.com>

piavW committed:
Updates usermodel association with trail_bookmarks

AlexSaxena committed:
Updates request-specs, rewrites model.

- Updates request-spec for bookmarks spec,
- Rewrites create action in bookmarks controller,
- Corrects typo in Users controller name

Co-Authored-By: Pia von Wachenfelt, <piavonwachenfelt@hotmail.com>

piavW committed:
Updates request test and controllers with actions

- Updates seeds with user_id
- Updates request test with get index request
- Updates bookmarks controllers actions index and create
- Updates user controller with show action for a users bookmarks

Co-Authored-By: Alex Saxena, <saxenaalex@gmail.com>

AlexSaxena committed:
Creates migrations
 - Migrations for Bookmarks, Trails and User associations

Co-Authored-By: Pia von Wachenfelt, <piavonwachenfelt@hotmail.com>

AlexSaxena committed:
Redoes associations
 - Redoes associations
 - Updates routes

Co-Authored-By: Pia von Wachenfelt, <piavonwachenfelt@hotmail.com>
